### ISSUE TITLE
syntax in .stylelintrc not working 

### ISSUE THREAD

**Issue by karolisgrinkevicius** [stylelint/stylelint#3290]
> Describe the issue. Is it a bug or a feature request (new rule, new option, etc.)?

Running `stylelint` without providing `--syntax scss` does not lint.

> Which rule, if any, is this issue related to?

All rules.

> What CSS is needed to reproduce this issue?

e.g.

```jsx
import styled from 'styled-components';

export const StyledElement = styled.div`
  display: block;

  &:after {
    display: none;
  }
`;
```

> What stylelint configuration is needed to reproduce this issue?

```json
{
  "processors": ["stylelint-processor-styled-components"],
  "extends": [
    "stylelint-config-recommended",
    "stylelint-config-styled-components"
  ],
  "syntax": "scss",
  "rules": {
    "selector-list-comma-newline-after": "always",
    "selector-pseudo-element-colon-notation": "double",
    "no-duplicate-selectors": null
  }
}
```

> Which version of stylelint are you using?

`9.2.0`

> How are you running stylelint: CLI, PostCSS plugin, Node API?

CLI with `stylelint --config .stylelintrc *.{ts,tsx}`

> Does your issue relate to non-standard syntax (e.g. SCSS, nesting, etc.)?

Yes, it's related to SCSS syntax.

> What did you expect to happen?

Error to be thrown complaining about `selector-pseudo-element-colon-notation` rule.

> What actually happened (e.g. what warnings or errors you are getting)?

`Exit 0`.

**Comment by evilebottnawi** [comment]
@karolisgrinkevicius why you need scss syntax for styled components?

**Comment by karolisgrinkevicius** [comment]
@evilebottnawi in case I use nested selectors in `styled-components` `stylelint` just doesn't work at all. If I prepend `--syntax scss` it actually starts working.

**Comment by jeddy3** [comment]
@karolisgrinkevicius This should be fixed in `9.2.1`.

Incidentally, there is no `syntax` property in the [stylelint configuration object](https://github.com/stylelint/stylelint/blob/master/docs/user-guide/configuration.md#the-configuration-object). There has never been. You can safely remove it. I started spotting it in people's configs a few months back, but I only recently tracked down the origin to an example within stylelint-processor-styled-components README. I submit [a PR](https://github.com/styled-components/stylelint-processor-styled-components/pull/171) to update their example.

**Comment by ViggoV** [comment]
@jeddy3 but then how are we supposed to use scss syntax with styled-components? If I run stylelint through the CLI with `--syntax scss` I get the expected outcome (support for `// comments` and nesting etc.) but without it, as is the case when using SublimeLinter, I get a bunch of unwanted `Uknown word` and `Unknown property value` errors.

**Comment by hudochenkov** [comment]
@ViggoV you most likely don't need `syntax` option, because it gets autodetected by stylelint. If you want to specify syntax for styled-components, then use `--syntax css-in-js`.

**Comment by ViggoV** [comment]
As mentioned using `--syntax scss` produces the expected outcome, but the issue was that I use stylelint through a text editor plugin rather than CLI. I did however resolve the issue as I realized SublimeLinter had an option to pass CLI arguments.
